block/manager.go

```diff
@@ -181,20 +181,6 @@ func getInitialState(ctx context.Context, genesis *RollkitGenesis, store store.S
 	if errors.Is(err, ds.ErrNotFound) {
 		logger.Info("No state found in store, initializing new state")
 
-		// Initialize genesis block explicitly
-		err = store.SaveBlockData(ctx,
-			&types.SignedHeader{Header: types.Header{
-				BaseHeader: types.BaseHeader{
-					Height: genesis.InitialHeight,
-					Time:   uint64(genesis.GenesisTime.UnixNano()),
-				}}},
-			&types.Data{},
-			&types.Signature{},
-		)
-		if err != nil {
-			return types.State{}, fmt.Errorf("failed to save genesis block: %w", err)
-		}
-
```

Comment:
we still want this no?

Reply:
The empty data is a bit strange thing. Unmarshalling emtpy slice doesn't panic/return error. I spend some time figuring this out ;)

Reply:
Isn't this needed to properly initialize go-header?